Add new CLI flag to exclude sources from elements

harrysarson opened the pull request:
For some buildstream elements generated by superflore we do not want to use the normal source. The way that buildstream includes work means it is not possible to overrite element sources.

This commit allows users of superflore-gen-bst to explicitly leave the `sources` of an element blank so that the sources can be specified by an include file.

tom--pollard commented on superflore/generators/buildstream/bst_element.py:
```diff
@@ -243,6 +279,8 @@ def get_element_text(self, distributor):
         if self.homepage:
             header += '# Homepage: ' + self.homepage + '\n'
         header += '# Source URI: ' + self.src_uri + '\n'
+        header += '# Git branch: ' + self.github_branch_url() + '\n'
```
Definitely think this addition to the comment metadata is worth-it, even if not using the exclude sources functionality

tom--pollard commented on superflore/generators/buildstream/bst_element.py:
```diff
@@ -157,6 +170,29 @@ def get_srcrev(self):
+
+    def github_commit_url(self):
+        """
+        Get a url for the comiit on github from a github archive SRC_URI, e.g.
```
s/comiit/commit

tom--pollard commented:
Nothing objectionable from my side on this one. I have suggested maybe emitting a warning/print of what it would have used for excluded packages in the past, but that's a nice to have if anything at this point

harrysarson commented:
I have pushed changes

1. `s/comiit/commit`
2. Added a `info` log when excluding a source
3. Fixed a bug where we would access `elements["sources"]` when sources excluded raising a `KeyError`.
